storage/innobase/srv/srv0start.cc

```diff
@@ -1860,9 +1855,6 @@ dberr_t srv_start(bool create_new_db)
 		    && !srv_read_only_mode) {
 			/* Drop partially created indexes. */
 			row_merge_drop_temp_indexes();
-			/* Rollback incomplete non-DDL transactions */
-			trx_rollback_is_active = true;
-			srv_thread_pool->submit_task(&rollback_all_recovered_task);
```

Comment:
I suspect that this needs to be preserved here in the MariaDB Embedded Server (`libmariadbd`). We may be missing proper test coverage of that, because `./mtr --embedded` tests do not allow the server to be restarted. This would have to be enclosed in `#ifndef HAVE_REPLICATION`.

Reply:
ok, I will keep the original code in embedded server.